common/src/main/java/io/netty/util/CharsetUtil.java

```diff
@@ -67,51 +68,113 @@
     public static Charset[] values() { return CHARSETS; }
 
     /**
-     * Returns a cached thread-local {@link CharsetEncoder} for the specified
-     * <tt>charset</tt>.
+     * @deprecated Use {@link #encoder(Charset)}.
      */
+    @Deprecated
     public static CharsetEncoder getEncoder(Charset charset) {
-        if (charset == null) {
-            throw new NullPointerException("charset");
-        }
+        return encoder(charset);
+    }
+
+    /**
+     * Returns a new {@link CharsetEncoder} for the <code>charset</code> with specified error actions.
+     *
+     * @param charset The specified charset
+     * @param malformedInputAction The encoder's action for malformed-input errors
+     * @param unmappableCharacterAction The encoder's action for unmappable-character errors
+     * @return The encoder for the specified <code>charset</code>
+     */
+    public static CharsetEncoder encoder(Charset charset, CodingErrorAction malformedInputAction,
+                                         CodingErrorAction unmappableCharacterAction) {
+        checkNotNull(charset, "charset");
```

Comment:
I guess I'd better write a compiler plugin that transforms `checkNotNull(a)` to `checkNotNull(a, "a")`